modules/utxo-lib/test/bitgo/psbt/paygoAddressProof.ts

```diff
@@ -0,0 +1,101 @@
+
+import { SignatureTargetType } from './Psbt';
+import { PSBT_PROPRIETARY_IDENTIFIER, UtxoPsbt } from 'modules/utxo-lib/src/bitgo';
+import { decodeProprietaryKey } from 'bip174/src/lib/proprietaryKeyVal';
+import { KeyValue } from 'bip174/src/lib/interfaces';
+
+const network = networks.bitcoin;
+const rootWalletKeys = getDefaultWalletKeys();
+
+const psbtInputs = inputScriptTypes.map((scriptType) => ({scriptType, value: BigInt(1000)}))
+const psbtOutputs = outputScriptTypes.map((scriptType) => ({ scriptType, value: BigInt(900)}))
+const sig = 'paygoaddresssig'
+
+function getTestPsbt(inputs: testutil.Input[], outputs: testutil.Output[], signed: SignatureTargetType) {
+    return testutil.constructPsbt(
+        inputs, outputs, network, rootWalletKeys, signed
+    )
+}
+
+function addPaygoAddressProofToPsbt(index: number, hasInput: boolean, hasOutput: boolean): UtxoPsbt {
+    // will update this function accordingly, just have this for now to help to simple testing
+    const psbt = getTestPsbt(hasInput ? psbtInputs : [], hasOutput ? psbtOutputs : [], hasInput && hasOutput ? 'unsigned' : 'fullsigned');
+    addPaygoAddressProof(psbt, index, Buffer.from(sig));
+    return psbt;
+}
+
+describe('addPaygoAddressProof function', () => {
+    function getPaygoProprietaryKey(proprietaryKeyVals: KeyValue[]) {
+        return proprietaryKeyVals.map(({key, value}) => {
+            return { key: decodeProprietaryKey(key), value };
+        }).filter((keyValue) => {
+            return keyValue.key.identifier === PSBT_PROPRIETARY_IDENTIFIER && keyValue.value === Buffer.from(sig)
+        });
+    }
+    it('should add PayGo Address proof to empty PSBT', () => {
+        const inputIndex = 0;
+        const psbt = getTestPsbt([], [], 'unsigned');
+        // better sig test replacement, i just have random string as test
+        addPaygoAddressProof(psbt, inputIndex, Buffer.from(sig));
+        const proprietaryKeyVals = psbt.data.globalMap.unknownKeyVals;
+        assert(proprietaryKeyVals)
+        // I assume that the proprietaryKeyVal should be properly added
+        // to the unknownKeyVals globalmap
+        const proofInPsbt = getPaygoProprietaryKey(proprietaryKeyVals)
+        assert(proofInPsbt.length === 1);
+    });
+
+    it('should add Paygo Adress Proof to non-empty PSBT', () => {
+        const inputIndex = 0;
+        const psbt = getTestPsbt(psbtInputs, psbtOutputs, 'fullsigned');
+        addPaygoAddressProof(psbt, inputIndex, Buffer.from(sig));
+        const proprietaryKeyVals = psbt.data.globalMap.unknownKeyVals;
+        assert(proprietaryKeyVals);
+        const proofInPsbt = getPaygoProprietaryKey(proprietaryKeyVals);
+        assert(proofInPsbt.length === 1);
+    });
+
+    // do we have an error check with the original addProprietaryKeyValToInput
+    // to see if there are duplicates of the same? if not I will add the test case here.
```

Comment:
make a test that has a proof on there already, and then you add a new one and make sure it gets substituted